apm-lambda-extension/extension/apm_server_transport.go

```diff
@@ -185,6 +185,12 @@ func (transport *ApmServerTransport) PostToApmServer(ctx context.Context, agentD
 		return fmt.Errorf("failed to read the response body after posting to the APM server")
 	}
 
+	if resp.StatusCode == http.StatusUnauthorized {
+		Log.Warnf("Authentication with the APM server failed: response status code: %d", resp.StatusCode)
+		Log.Debugf("APM server response body: %v", string(body))
+		return nil
```

Comment:
**question:** should this return an error or is it too disruptive ?

Reply:
As you pointed out, #225  is a generalization where also the health state should be adapted for various scenarios. This will require a slight rework of the [healthstate handling](https://github.com/elastic/apm-aws-lambda/pull/228/files#diff-84dece7b8082ce96bbb9405a021562cc58b344c8558ff8d38816f564c372c9e5R208-R235). I don't mind if you want to merge this improvement separately as is now and then create a follow up PR or add the health state handling and checking if there are other cases that need to be handled in this PR.

Reply:
That's fine, just please remove this from the description and only close the issue when the health state is also fixed. 

> Closes https://github.com/elastic/apm-aws-lambda/issues/205

Reply:
Done!